### PR TITLE
18449 - GP/SP registration modify

### PR DESCRIPTION
**Pull request by tshyun24**
*Issue #:* /bcgov/entity#18449

*Description of changes:* Modify GP/SP registration as new UX requirement

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-582-sgk75ejj.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-582-sgk75ejj.web.app

File: src/components/common/RegAddEditOrgPerson.vue [hunk 1]
```diff
@@ -102,104 +116,12 @@
               </div>
             </article>
 
-            <!-- Business or Corporation -->
+            <!-- Business or Corporation Look up-->
             <template v-if="isOrg">
-              <!-- Business or Corporation Unregistered in B.C. -->
```
Comment (tshyun24) [inline src/components/common/RegAddEditOrgPerson.vue:107]: remove "business or corporation unregistered in B.C." since we do not want this section now, only "existing B.C. business" now
image

File: src/components/common/RegAddEditOrgPerson.vue [hunk 2]
```diff
@@ -215,46 +137,27 @@
...
                 <BusinessLookup
                   :showErrors="enableRules"
                   :businessLookup="inProgressBusinessLookup"
                   :BusinessLookupServices="BusinessLookupServices"
+                  :label="businessLookupLabel"
                   @setBusiness="updateBusinessDetails($event)"
                   @undoBusiness="resetBusinessDetails()"
                 />
+
+                <HelpContactUs class="mt-0.25" />
```
Comment (tshyun24) [inline src/components/common/RegAddEditOrgPerson.vue:160]: to meet the 30px space above and 40px space below requirement
image

File: src/components/common/RegistriesContactInfo.vue [hunk 3]
```diff
@@ -22,22 +22,9 @@
       </v-icon>
       <span class="px-2">Victoria Office:</span>
       <a
-        href="tel:+1-250-387-7848"
+        href="tel:+1-250-370-1033"
```
Comment (tshyun24) [inline src/components/common/RegistriesContactInfo.vue:25]: change the help number and remove email
image

File: tests/unit/RegAddEditOrgPerson.spec.ts [hunk 4]
```diff
@@ -363,44 +363,18 @@ describe('Registration Add/Edit Org/Person component', () => {
     wrapper.destroy()
   })
 
-  it('displays form data for proprietor-org (SP) - manual add', async () => {
```
Comment (tshyun24) [inline tests/unit/RegAddEditOrgPerson.spec.ts:366]: remove it because we do not accept manually added SP now

File: tests/unit/RegAddEditOrgPerson.spec.ts [hunk 5]
```diff
@@ -363,44 +363,25 @@ describe('Registration Add/Edit Org/Person component', () => {
...
+    expect(wrapper.find('.business-lookup-article label').text()).toContain('Business')
+    expect(wrapper.findAll('.business-lookup-article p').at(0).text()).toContain('Enter an existing B.C. business as the proprietor')
 
     wrapper.destroy()
   })
 
   it('displays form data for proprietor-org (SP) - edit', async () => {
     const wrapper = createComponent(validProprietorOrg, 0, null)
 
-    await wrapper.find('.lookup-toggle').trigger('click')
```
Comment (tshyun24) [inline tests/unit/RegAddEditOrgPerson.spec.ts:390]: remove `.lookup-toggle` because we do not use it anymore, and no confirm checkout, cannot let people to input their organzation name

**Comment by tshyun24** [comment]
Test for SP
NR 4763379

Test for GP
NR 9187663

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-582-sgk75ejj.web.app

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "business-create-ui",
-  "version": "5.5.8",
+  "version": "5.5.9",
```
Comment (severinbeauvais) [inline package.json:3]: Bump this to 5.6.x to match my recent commits please.

File: src/components/common/RegAddEditOrgPerson.vue [hunk 7]
```diff
@@ -21,8 +21,22 @@
             class="add-org-header"
             :class="{'error-text': !addPersonOrgFormValid}"
           >
-            <span v-if="isNaN(activeIndex)">Add Business or Corporation</span>
-            <span v-else>Edit Business or Corporation</span>
+            <span v-if="isNaN(activeIndex)">
```
Comment (severinbeauvais) [inline src/components/common/RegAddEditOrgPerson.vue:24]: Instead of having a span wrapping other spans (and not adding styles or anything), use the Vue `template` element here.

File: src/components/common/RegistriesContactInfo.vue
Comment (severinbeauvais) [inline src/components/common/RegistriesContactInfo.vue:1]: Check where else this component is used. If it's only for SP/GP registrations then the new phone numbers are probably correct. But these might be the wrong phone numbers for other filings (IAs, dissolutions, restorations).

File: src/components/common/RegAddEditOrgPerson.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue:160]: Where did you find this classname?

File: src/components/common/RegAddEditOrgPerson.vue [hunk 8]
```diff
@@ -215,46 +137,27 @@
...
+                    Enter an existing B.C. business as a partner.
                   </p>
+
                 </div>
 
                 <BusinessLookup
                   :showErrors="enableRules"
                   :businessLookup="inProgressBusinessLookup"
                   :BusinessLookupServices="BusinessLookupServices"
+                  :label="businessLookupLabel"
```
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue:155]: Why not hard-code this instead of using a variable/property?

File: tests/unit/RegAddEditOrgPerson.spec.ts (hunk 4 above)
Comment (severinbeauvais) [reply tests/unit/RegAddEditOrgPerson.spec.ts:366]: As above, if we are only hiding this functionality for now then I think you should disable this unit test instead of deleting it. (Use `it.skip`.)

Comment (severinbeauvais) [reply tests/unit/RegAddEditOrgPerson.spec.ts:366]: Ditto for other unit test changes below.

File: src/components/common/RegAddEditOrgPerson.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue:107]: Also, I see in the uxpin, "Don't make changes for Registries stadd. Make the changes for SBC staff and clients."

File: src/components/common/RegAddEditOrgPerson.vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (tshyun24) [reply src/components/common/RegAddEditOrgPerson.vue:155]: Because we are calling `BusinessLookup` from `'@bcrs-shared-components/business-lookup'`, and we set the previous label as a prop cause we do not want to effect other UIs.

File: src/components/common/RegAddEditOrgPerson.vue (hunk 2 above)
Comment (tshyun24) [reply src/components/common/RegAddEditOrgPerson.vue:160]: I find it from (https://vuetifyjs.com/en/styles/spacing/#how-it-works), I try different spacing to make sure it is 30px above and 40px below
image
image

File: src/components/common/RegAddEditOrgPerson.vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue:155]: Are you saying this instance of BusinessLookup (in this Vue file) is used by other filings or other UIs?

[the remark this answers is not in the document]
Comment (tshyun24) [reply src/components/common/RegAddEditOrgPerson.vue:155]: I think it refers to what you said here? If I don't misunderstand
image

File: src/components/common/RegAddEditOrgPerson.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue:160]: I think that classname doesn't exist.

Look, it doesn't have a definition in the DOM:

image

In fact, there isn't a Vuetify helper class for "30px". We either use 30px or 32px (factors of 4). Ask Yui what to do here.

(Example: mt-3 means 3/4 rem = 12 px.)

File: src/components/common/RegAddEditOrgPerson.vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue:155]: The BusinessLookup component is used in multiple places, but this specific instance of it (in RegAddEditOrgPerson.vue) is used only by firm registrations, so if you don't need different labels here then you can hard-code it.

(One exception to hard-coding a static string is if it's too long to put in the template; then we sometimes declare a variable for it in the script block.)

[the remark this answers is not in the document]
Comment (tshyun24) [reply src/components/common/RegAddEditOrgPerson.vue:155]: Got it!! Thanks Severin, sorry for misunderstanding "hard-code" at here😥

File: src/components/common/RegistriesContactInfo.vue
Comment (tshyun24) [reply src/components/common/RegistriesContactInfo.vue:1]: This component have been used for lots of other place, so in that case, should I create a contact info component only for SP/GP registrations and insert at here or?
image

Comment (severinbeauvais) [reply src/components/common/RegistriesContactInfo.vue:1]: The easiest thing to track would be to leave this component alone and create a new one.

Be aware that we have been confused by different phone numbers in the past -- different teams look after different entity types or filings.

Chat with our UI team (eg, Yui) to come up with a good name for the new contact info component -- are these phone numbers specific to firm (SP/GP) support? Maybe something like "FirmContactInfo.vue"?

Comment (tshyun24) [reply src/components/common/RegistriesContactInfo.vue:1]: confirmed with Yui. So I will leave this alone and create a new one
image

**Comment by severinbeauvais** [comment]
Shaoyun, please fix the lint and unit test errors, and also update the app version (which was lost when you rebased). Thanks.

**Comment by tshyun24** [comment]
> Shaoyun, please fix the lint and unit test errors, and also update the app version (which was lost when you rebased). Thanks.

Hii Severin, I just push a new commit, this is what I mainly changed from last week, having the old UI page for Registries Staff only and new UI page for SBC Staff and clients

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-582-sgk75ejj.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-582-sgk75ejj.web.app

File: src/components/Registration/HelpContactUs.vue [hunk 9]
```diff
@@ -31,11 +31,19 @@
...
-        </p>
+        <div v-if="isTypeFirm">
+          <p class="my-4">
+            If you require further assistance adding a business, please contact us.
+          </p>
+          <FirmContactInfo />
+        </div>
 
-        <RegistriesContactInfo />
+        <div v-else>
```
Comment (severinbeauvais) [inline src/components/Registration/HelpContactUs.vue:41]: This "else" case will never be hit, because this component is used for registrations, which are only for firms.

File: src/components/common/RegAddEditOrgPerson.vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue]: LGTM

File: src/components/common/RegAddEditOrgPerson.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue]: LGTM

File: src/components/common/RegistriesContactInfo.vue (hunk 3 above)
Comment (severinbeauvais) [reply src/components/common/RegistriesContactInfo.vue]: This is an obsolete conversation -- there is now a new component for firm contact info. Resolving.

File: tests/unit/RegAddEditOrgPerson.spec.ts (hunk 4 above)
Comment (severinbeauvais) [reply tests/unit/RegAddEditOrgPerson.spec.ts:366]: LGTM

File: package.json (hunk 6 above)
Comment (severinbeauvais) [reply package.json]: LGTM

File: src/components/common/RegAddEditOrgPerson.vue (hunk 7 above)
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue]: LGTM

File: src/components/common/RegistriesContactInfo.vue
Comment (severinbeauvais) [reply src/components/common/RegistriesContactInfo.vue:1]: This component is no longer changed. Resolving.

**Review by severinbeauvais — APPROVED** [review]
Please get a couple more reviews before merging.

File: src/components/common/FirmContactInfo.vue [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+
+.contact-container {
+  font-size: $px-15;
+  color: $gray7;
+  display: flex;
+
+  .contact-icon {
```
Comment (JazzarKarim) [inline src/components/common/FirmContactInfo.vue:47]: Please Shaoyun if you can fix the indentation here and the class below.

Comment (severinbeauvais) [reply src/components/common/FirmContactInfo.vue:47]: Please also stop importing the obsolete component, and the store getter and imports.

File: src/components/Registration/HelpContactUs.vue [hunk 11]
```diff
@@ -51,14 +51,19 @@
...
 @Component({
   components: {
-    RegistriesContactInfo
+    RegistriesContactInfo,
+    FirmContactInfo
   }
 })
 export default class HelpContactUs extends Vue {
   helpToggle = false
+  @Getter(useStore) isTypeFirm!: boolean
```
Comment (JazzarKarim) [inline src/components/Registration/HelpContactUs.vue:66]: It's fine now but we usually have the getters first then follow by local variables. 

For example:
https://github.com/bcgov/business-create-ui/blob/700a8b48d3cc4570d2b060f4be0ddb6c53e273bf/src/views/Registration/RegistrationDefineBusiness.vue#L180

File: src/components/common/FirmContactInfo.vue (hunk 10 above)
Comment (tshyun24) [reply src/components/common/FirmContactInfo.vue:47]: > Please Shaoyun if you can fix the indentation here and the class below.

image
I think the indentation should be that, `contact-icon` and `contact-value` class are under `class-container`

File: src/components/Registration/HelpContactUs.vue (hunk 11 above)
Comment (JazzarKarim) [reply src/components/Registration/HelpContactUs.vue:66]: Actually, you can remove this line Shaoyun please because you're not using the getter in this file.

File: tests/unit/HelpContactUs.spec.ts [hunk 12]
```diff
@@ -31,10 +31,10 @@ describe('HelpBusinessNumber', () => {
     expect(wrapper.find('.help-header h2').text()).toBe('Contact BC Registries')
 
     // verify paragraph
-    expect(wrapper.find('p').text()).toContain('If you require further assistance')
+    expect(wrapper.find('p').text()).toContain('If you require further assistance adding a business, please contact us.')
```
Comment (JazzarKarim) [inline tests/unit/HelpContactUs.spec.ts:34]: Can you please make this to use two lines? We usually don't go over the length of 120 characters. I see GitHub complaining about that. 

If you run `npm run lint`, you can see all the warnings in regards to the maximum length. Please Shaoyun if you can fix it here and below.

File: src/components/common/FirmContactInfo.vue (hunk 10 above)
Comment (tshyun24) [reply src/components/common/FirmContactInfo.vue:47]: > Please also stop importing the obsolete component, and the store getter and imports.

Do you mean removing this part?
image

Comment (JazzarKarim) [reply src/components/common/FirmContactInfo.vue:47]: That's fine then. I was thinking you could make the classes separate (not nested) but in your case it also works. Resolving, thanks.

**Review by JazzarKarim — APPROVED** [review]
Aside from my couple comments, looks good. Great job!

File: src/components/common/FirmContactInfo.vue [hunk 13]
```diff
@@ -29,29 +29,22 @@
   </ul>
 </template>
 
-<script lang="ts">
-import { Component, Vue } from 'vue-property-decorator'
```
Comment (JazzarKarim) [inline src/components/common/FirmContactInfo.vue]: Please revert this change Shaoyun! The component won't work without it here.

File: tests/unit/HelpContactUs.spec.ts (hunk 12 above)
Comment (JazzarKarim) [reply tests/unit/HelpContactUs.spec.ts]: Fixed, thanks. Resolving.

File: src/components/Registration/HelpContactUs.vue (hunk 11 above)
Comment (JazzarKarim) [reply src/components/Registration/HelpContactUs.vue]: Fixed, thanks. Resolving.

File: src/components/common/FirmContactInfo.vue (hunk 13 above)
Comment (JazzarKarim) [reply src/components/common/FirmContactInfo.vue]: Fixed, thanks. Resolving.

**Review by PaulGarewal — APPROVED** [review]
If all comments are resolved, LGTM 👍

File: src/components/Registration/HelpContactUs.vue [hunk 14]
```diff
@@ -51,10 +51,12 @@
 <script lang="ts">
 import { Component, Vue } from 'vue-property-decorator'
 import RegistriesContactInfo from '@/components/common/RegistriesContactInfo.vue'
+import FirmContactInfo from '@/components/common/FirmContactInfo.vue'
 
 @Component({
   components: {
-    RegistriesContactInfo
+    RegistriesContactInfo,
```
Comment (severinbeauvais) [inline src/components/Registration/HelpContactUs.vue:58]: Please remove this obsolete component (and its import).

File: src/components/Registration/HelpContactUs.vue (hunk 9 above)
Comment (severinbeauvais) [reply src/components/Registration/HelpContactUs.vue]: LGTM

File: src/components/Registration/HelpContactUs.vue (hunk 14 above)
Comment (severinbeauvais) [reply src/components/Registration/HelpContactUs.vue]: LGTM

**Review by jamespaologarcia — APPROVED** [review]
LGTM

File: src/components/common/RegAddEditOrgPerson.vue [hunk 15]
```diff
@@ -21,8 +21,30 @@
...
+              </template>
+
+              <template v-else>
+                <template v-if="isTypePartnership">
+                  Edit a Business as a Partner
+                </template>
+                <template v-else>
+                  Edit a Business as the Proprietor
+                </template>
+              </template>
```
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/components/common/RegAddEditOrgPerson.vue:46]: James and I discussed this, and he's right, but I think leaving the code like this is also acceptable as it's clear what's going on.
